jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -214,26 +245,56 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
             nddata = spectral_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=NDDataArray
             )
-            uncertainties = uncert_cube.get_subset_object(
-                subset_id=self.aperture.selected, cls=StdDevUncertainty
+            if uncert_cube:
+                uncertainties = uncert_cube.get_subset_object(
+                    subset_id=self.aperture.selected, cls=StdDevUncertainty
+                )
+            # Exact slice mask of cone or cylindrical aperture through the cube. `shape_mask` is
+            # a 3D array with fractions of each pixel within an aperture at each
+            # wavelength, on the range [0, 1].
+            if self.function_selected.lower() in ['min', 'max'] and self.wavelength_dependent:
+                shape_mask = self.cone_aperture().astype(int)
+            elif self.function_selected.lower() in ['min', 'max']:
+                shape_mask = self.cylindrical_aperture().astype(int)
+            elif self.wavelength_dependent:
+                shape_mask = self.cone_aperture()
+            else:
+                shape_mask = self.cylindrical_aperture()
```

Comment:
Can be simplified?
```suggestion
            if self.wavelength_dependent:
                shape_mask = self.cone_aperture()
            else:
                shape_mask = self.cylindrical_aperture()

            if self.function_selected.lower() in ('min', 'max'):
                shape_mask = shape_mask.astype(int)
```

Reply:
Actually, lots of duplicated code between `cone_aperture()` and `cylindrical_aperture()`. I wonder if you only need one `get_aperture()` method and have the `self.wavelength_dependent` check inside that method instead.

Reply:
Done.